platform/broadcom/sonic-platform-modules-inventec/d6356/modules/i2c-mux-pca9541.c

```diff
@@ -505,13 +505,13 @@ static int pca9541_probe(struct i2c_client *client,
 	 * We have to lock the adapter before releasing the bus.
 	 */
 	if (detect_id == 0) {
-		i2c_lock_adapter(adap);
+		i2c_lock_bus(adap, I2C_LOCK_ROOT_ADAPTER);
 		pca9541_release_bus(client);
-		i2c_unlock_adapter(adap);
+		i2c_unlock_bus(adap, I2C_LOCK_ROOT_ADAPTER);
 	} else {
-		i2c_lock_adapter(adap);
+		i2c_lock_bus(adap, I2C_LOCK_ROOT_ADAPTER);
 		pca9641_release_bus(client);
-		i2c_unlock_adapter(adap);
+		i2c_unlock_bus(adap, I2C_LOCK_ROOT_ADAPTER);
```

Comment:
i do not know how driver got into sonic repo, we should use standard linux driver for 9541.

Reply:
This part is modified  after previous feedback, please help to review, thank you!